src/Actions/OutputsDslQueryAction.php

```diff
@@ -153,28 +159,76 @@ public function build(array $config): array
         ];
     }
 
+    protected function processGroup(array $group): ?array
+    {
+        $groupConjunction = strtoupper($group['conjunction'] ?? 'AND');
+        $groupKey = $groupConjunction === 'OR' ? 'should' : 'must';
+        $conditions = [];
+
+        foreach ($group['conditions'] as $condition) {
+            if (isset($condition['type']) && $condition['type'] === 'group') {
+                $nestedGroupClause = $this->processGroup($condition);
+                if ($nestedGroupClause !== null) {
+                    $conditions[] = $nestedGroupClause;
+                }
+            } else {
+                if (isset($condition['operator'])) {
+                    $conditions[] = $this->mapCondition($condition);
+                }
+            }
+        }
+
+        if (empty($conditions)) {
+            return null;
+        }
+
+        if (count($conditions) === 1) {
+            return $conditions[0];
+        }
+
+        return ['bool' => [$groupKey => $conditions]];
```

Comment:
Maybe use laravel collections here to make it a little more readable? There's a lot of nested ifs

totally untested but maybe something like:
```php
$conditions = collect($group['conditions'] ?? [])
    ->map(function ($condition) {
        if ($condition['type']  === 'group') {
            return $this->processGroup($condition);
        }

        if (isset($condition['operator'])) {
            return $this->mapCondition($condition);
        }

        return null;
    })
    ->filter()
    ->values();
```